pkg/ucp/credentials/azure.go

```diff
@@ -69,10 +69,17 @@ func (p *AzureCredentialProvider) Fetch(ctx context.Context, planeName, name str
 		case *ucpapi.InternalCredentialStorageProperties:
 			storage = c
 		default:
-			return nil, errors.New("invalid AzureServicePrincipalProperties")
+			return nil, errors.New("invalid InternalCredentialStorageProperties")
```

Comment:
Is this error customer facing?

Reply:
maybe, it looks like it can be returned from the credential API call